Source/NSDockTile.m

```diff
@@ -90,13 +111,104 @@ - (NSString *) badgeLabel
 
 - (void) setBadgeLabel: (NSString *)label
 {
+  NSImage *tempImage;
+
   ASSIGNCOPY(_badgeLabel, label);
+
+  tempImage = [[NSImage alloc] initWithSize: [_appIconImage size]];
+  [tempImage addRepresentation: _imageRep];
+  [NSApp setApplicationIconImage: tempImage];
+  RELEASE(tempImage);
 }
 
 - (void) display
 {
   [_contentView setNeedsDisplay: YES];
 }
 
-@end
+- (void)draw
+{
+  [_appIconImage compositeToPoint: NSZeroPoint operation: NSCompositeCopy];
+
+  if (_showsApplicationBadge && _badgeLabel)
+    {
+      NSMutableDictionary *attrs;
+      NSPoint textLocation;
+      NSRect discRect;
+      NSSize discSize;
+      NSSize textSize;
+      int pad;
+      NSBezierPath *p;
+      NSPoint point;
+      CGFloat radius;
+      NSSize imageSize;
+      NSColor *badgeBackColor;
+      NSColor *badgeDecorationColor;
+      NSColor *badgeTextColor;
+      NSString *displayString;
+
+      badgeBackColor = [NSColor redColor];
+      badgeDecorationColor = [NSColor lightGrayColor];
+      badgeTextColor = [NSColor whiteColor];
+
+      imageSize = [_appIconImage size];
+
+      displayString = _badgeLabel;
+      if ([_badgeLabel length] > 5)
+	displayString = [NSString stringWithFormat:@"%@\u2026%@", [_badgeLabel substringToIndex:2], [_badgeLabel substringFromIndex:[_badgeLabel  length]-2]];
```

Comment:
Our coding standard suggests to use {} even for one line if statements. Also a few spaces are still missing in this line (at least four).

Reply:
I counted three... I don't like the {} for one liners, I'm decidedly against it, but GUI is your code.. so should be compliant to your liking now

Reply:
> I counted three... I don't like the {} for one liners, I'm decidedly against it, but GUI is your code.. so should be compliant to your liking now

It is part of the GNU coding standard.  I, personally, think it makes things clearer and easier to follow, but to each their own.